src/Microsoft.Framework.PackageManager/Algorithm/DFPostOrderWalker.cs

```diff
@@ -0,0 +1,32 @@
+﻿// Copyright (c) Microsoft Open Technologies, Inc. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Linq;
+using System.Collections.Generic;
+
+namespace Microsoft.Framework.PackageManager.Algorithm
+{
+    /// <summary>
+    /// Walk the graph in depth-first post order 
+    /// </summary>
+    public class DFPostOrderWalker<T, K>
```

Comment:
Thought about that but the name appears a little bit long. So here's my thought, I'd like to have a set of common used algorithms and data structures place together. There name can follow the convention that `DF` stands for `Depth First` and `BF` stands for `Breath First`. It helps to shorten the name without hurt the readability to much.